src/shader_recompiler/frontend/translate/data_share.cpp

```diff
@@ -110,6 +116,50 @@ void Translator::DS_WRITE(int bit_size, bool is_signed, bool is_pair, const GcnI
     }
 }
 
+void Translator::DS_MAX_U32(const GcnInst& inst) {
+    const IR::U32 addr{GetSrc(inst.src[0])};
+    const IR::U32 data{GetSrc(inst.src[1])};
+    const IR::U32 offset = ir.Imm32(u32(inst.control.ds.offset0));
+    const IR::U32 addr_offset = ir.IAdd(addr, offset);
+
+    const IR::U32 old_value = IR::U32(ir.LoadShared(32, false, addr_offset));
+    const IR::U32 new_value = ir.UMax(old_value, data);
+    ir.WriteShared(32, new_value, addr_offset);
+
+    if (inst.dst[0].type != ScalarType::Undefined) {
+        SetDst(inst.dst[0], old_value);
+    }
+}
+
+void Translator::DS_MIN_U32(const GcnInst& inst) {
+    const IR::U32 addr{GetSrc(inst.src[0])};
+    const IR::U32 data{GetSrc(inst.src[1])};
+    const IR::U32 offset = ir.Imm32(u32(inst.control.ds.offset0));
+    const IR::U32 addr_offset = ir.IAdd(addr, offset);
+
+    const IR::U32 old_value = IR::U32(ir.LoadShared(32, false, addr_offset));
+    const IR::U32 new_value = ir.UMin(old_value, data);
+    ir.WriteShared(32, new_value, addr_offset);
+
+    if (inst.dst[0].type != ScalarType::Undefined) {
+        SetDst(inst.dst[0], old_value);
+    }
+}
+
+void Translator::DS_ADD_U32(const GcnInst& inst) {
+    const IR::U32 addr{GetSrc(inst.src[0])};
+    const IR::U32 data{GetSrc(inst.src[1])};
+    const IR::U32 offset = ir.Imm32(u32(inst.control.ds.offset0));
+    const IR::U32 addr_offset = ir.IAdd(addr, offset);
```

Comment:
This also need a special IR instruction (can call it SharedAtomicIAdd32). Also I don't think its necessary to align down address as guest should pass dword aligned values when reading dwords